src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -345,6 +347,91 @@ static bool is_input_uniform_constant(shared_ptr<Node> op,
     return rc;
 }
 
+//`simplify_gather`, optimizes gather if Gather is gathering the
+// whole input tensor
+static bool simplify_gather(std::shared_ptr<Node> node)
+{
+    auto gather = as_type_ptr<op::v1::Gather>(node);
+
+    if (gather)
+    {
+        // check if we are gathering the whole input
+        auto data = node->get_argument(0);
+        auto indices = node->get_argument(1);
+
+        // we need to know data and indices shape to infer if gather is Nop
+        if (data->get_output_partial_shape(0).is_dynamic() ||
+            indices->get_output_partial_shape(0).is_dynamic())
+        {
+            return false;
+        }
+        // if rank of data and gather output dont match, we will skip
+        if (data->get_shape().size() != node->get_shape().size())
+        {
+            return false;
+        }
+
+        auto axis = gather->get_axis();
+
+        if (axis == std::numeric_limits<int64_t>::max())
+        {
+            NGRAPH_DEBUG << "axis value not set";
+            return false;
+        }
+
+        // lamba function to remove the gather op and rewire the
+        // input data of gather to its users
+        auto replace_gather = [&]() {
+            remove_node_update_name(node, node->get_argument(0));
```

Comment:
return the result of this function. also, why not directly call this func instead of lambda